server.go

```diff
@@ -245,10 +245,22 @@ func (s *Server) register(sd *ServiceDesc, ss interface{}) {
 	s.m[sd.ServiceName] = srv
 }
 
-// ServiceInfo contains method names and metadata for a service.
+// StreamInfo contains information about a streaming RPC.
+type StreamInfo struct {
+	// Name is the RPC name only, without the service name or package name.
+	Name string
+	// IsClientStream indicates whether the RPC is a client streaming RPC.
+	IsClientStream bool
+	// IsServerStream indicates whether the RPC is a server streaming RPC.
+	IsServerStream bool
+}
+
+// ServiceInfo contains unary RPC names, streaming RPC infos and metadata for a service.
 type ServiceInfo struct {
-	// Methods are method names only, without the service name or package name.
+	// Methods are unary RPC names only, without the service name or package name.
 	Methods []string
+	// Streams are streaming RPC names and streaming types.
+	Streams []*StreamInfo
```

Comment:
merge the info for unary and stremaing like

```
Unary []Info
Stream []Info
```
For unary rpc, both IsClientStream and IsServerStream are false.